urlead5353366_Zeldon_BigData_Class_tgz-pJ8-LocalTestsJ8/src/com/beust/jcommander/internal/JDK6Console.java

```diff
@@ -0,0 +1,45 @@
+package com.beust.jcommander.internal;
+
+import com.beust.jcommander.ParameterException;
+
+import java.io.PrintWriter;
+import java.lang.reflect.Method;
+
+public class JDK6Console implements Console {
+
+  private Object console;
+
+  private PrintWriter writer;
+
+  public JDK6Console(Object console) throws Exception {
+    this.console = console;
+    Method writerMethod = console.getClass().getDeclaredMethod("writer", new Class<?>[0]);
+    writer = (PrintWriter) writerMethod.invoke(console, new Object[0]);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Catch `InvocationTargetException` explicitly when you call `Method.invoke()`. Use `Throwable.getCause()` or `getTargetException()` in the catch handler to access more information about the underlying exception. See the following diffs on GitHub that make this code change: [diff1](https://github.com/EsfingeFramework/gamification/commit/20792047e6c69457cca03e877f2ada578188f32a#diff-6808c12fe9ddaea042db84f95dd5ba78adab99d0cb18bb4d85916a8723530768L21), [diff2](https://github.com/cucumber-attic/cuke4duke/commit/f4821d400cf2590126f1298a8dff9f7a33d45dc5#diff-80adaffc20c0e6e7d2e42312386ffe1ddc10213bf5b8e8919895fd85d666c8deL34)

Learn more: [InvocationTargetException in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/InvocationTargetException.html)